daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/XMLTextInfosetOutputter.scala

```diff
@@ -31,12 +31,15 @@ import org.apache.daffodil.util.Indentable
  * @param writer The writer to write the XML text to
  * @param pretty Whether or to enable pretty printing. Set to true, XML
  *               elements are indented and newlines are inserted.
+ * @param xmlOutputStyle Determine whether to wrap values of elements of type
+ *        xs:string in CDATA chunks in order to preserve whitespace.
  */
-class XMLTextInfosetOutputter private (writer: java.io.Writer, pretty: Boolean)
+class XMLTextInfosetOutputter private (writer: java.io.Writer, pretty: Boolean, xmlOutputStyle: XMLOutputStyle.Type)
   extends InfosetOutputter with Indentable with XMLInfosetOutputter {
 
-  def this(os: java.io.OutputStream, pretty: Boolean) = {
-    this(new java.io.OutputStreamWriter(os, StandardCharsets.UTF_8), pretty)
+  def this(os: java.io.OutputStream, pretty: Boolean, xmlOutputStyle: XMLOutputStyle.Type = XMLOutputStyle.Standard) = {
+    this(new java.io.OutputStreamWriter(os, StandardCharsets.UTF_8), pretty, xmlOutputStyle)
+    Assert.usage(xmlOutputStyle != null, "Null is not a valid value for parameter xmlOutputStyle")
```

Comment:
What's our standard for validating API constructor/function argumens? Users could pass in null to a bunch of our APIs and they would end up with a NullPointerException. This particular parameter doesn't feel any different than any other parameters, so should we even be checking null here? Or should we open a bug to validate non-nullness for all parameters?